code/extensions/che-commands/src/taskProvider.ts

```diff
@@ -75,9 +75,38 @@ export class DevfileTaskProvider implements vscode.TaskProvider {
 		};
 
 		const execution = new vscode.CustomExecution(async (): Promise<vscode.Pseudoterminal> => {
-			return this.terminalExtAPI.getMachineExecPTY(component, command, expandEnvVariables(workdir));
+			return this.terminalExtAPI.getMachineExecPTY(component, resolveEnvVariablesForCommand(command, env), expandEnvVariables(workdir));
```

Comment:
@azatsarynnyy 
I would like to know your opinion here:
- is it enough to resolve such variables on this step?
 so we don't set such env variable to the corresponding container, we resolve `command` when `TaskProvider` creates tasks
 or
- the issue assumes that such variable should be set to the corresponding container and available from a terminal as well?

Reply:
[Devfile spec v. 2.2.0](https://devfile.io/docs/2.2.0/devfile-schema#commands-exec-env) says that a command's env. variables list is
> `Optional list of environment variables that have to be set before running the command`

I understand it as such variables should be set only for a specific command execution.